### ISSUE TITLE
[Feature] Forward the axis object to formatter functions

### ISSUE THREAD

**Issue by ob-ARC** [apache/echarts#16429]
### What problem does this feature solve?

In a formatter callback function, it is currently not possible to change the returned format depending on the axis' state, since it is not known for which axis the formatting is requested. Use case: Only show mSecs on a date/time axis if the axis is zoomed in enough (i.e. the range is below a certain threshold).

### What does the proposed API look like?

The solution would simply be to let _getFormattedLabel_ forward "this" (= the axis to label) through _leveledformat_ to the user's callback function.

**Comment by Ovilia** [comment]
Do you mean time axis?

**Comment by ob-ARC** [comment]
Yes, my use case was a time axis. But I guess this feature could be useful for other axis types as well?

**Comment by Ovilia** [comment]
I think maybe [listening to dataZoom changes](https://echarts.apache.org/en/api.html#events.datazoom) and call setOption may be a better idea. Even if the axis is provided in the label formatter, it will not be called when dataZoom changes.

**Comment by ob-ARC** [comment]
OK, but that would still leave me the problem that in the formatter callback itself, I would neither know which axis is being labelled, nor what the current state of the axis is. Or do you mean I should set a different callback function to the chart's option each time dataZoom changes?

**Comment by sjcobb** [comment]
I definitely agree that this would offer a lot of flexibility, I've tried using 'time' axis but ran into a few issues (#14305, #16685) where it made sense to change back to 'category'

**Comment by ob-ARC** [comment]
I am once again confronted with the issue that I do not know which axis is being labelled in the formatter function. Is there any news/workaround for this?
